From e9c2a806e2d8ebd792e751b28abe698de217cbbb Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 25 May 2014 20:48:33 +0300
Subject: [PATCH 45/46] ENET_MDIO should be open drain

Etheret MDIO signal should be open drain by spec (pulled up on board).
This caused some marginal issues on some micro SOMs that the artifact was link up/down few times.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/plat-mxc/include/mach/iomux-mx6dl.h |    6 +++++-
 arch/arm/plat-mxc/include/mach/iomux-mx6q.h  |    6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h b/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
index 59f8399..68c4715 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
@@ -58,6 +58,10 @@
 		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |		\
 		PAD_CTL_DSE_40ohm   | PAD_CTL_HYS)
 
+#define MX6DL_ENET_OD_PAD_CTRL (PAD_CTL_PKE | PAD_CTL_PUE  |		\
+		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |		\
+		PAD_CTL_DSE_40ohm   | PAD_CTL_HYS | PAD_CTL_ODE)
+
 #define MX6DL_ENET_REF_CLK_PAD_CTRL     (PAD_CTL_PUE |  \
 		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |       \
 		PAD_CTL_DSE_40ohm | PAD_CTL_SRE_FAST)
@@ -2258,7 +2262,7 @@
 		IOMUX_PAD(0x05B8, 0x01E8, 7, 0x0000, 0, NO_PAD_CTRL)
 
 #define MX6DL_PAD_ENET_MDIO__ENET_MDIO                                         \
-		IOMUX_PAD(0x05BC, 0x01EC, 1, 0x0810, 0, MX6DL_ENET_PAD_CTRL)
+		IOMUX_PAD(0x05BC, 0x01EC, 1, 0x0810, 0, MX6DL_ENET_OD_PAD_CTRL)
 #define MX6DL_PAD_ENET_MDIO__ESAI1_SCKR                                        \
 		IOMUX_PAD(0x05BC, 0x01EC, 2, 0x083C, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_ENET_MDIO__SDMA_DEBUG_BUS_DEVICE_3                           \
diff --git a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
index 16cd87f..e3eca91 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
@@ -57,6 +57,10 @@
 		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |	\
 		PAD_CTL_DSE_40ohm | PAD_CTL_HYS)
 
+#define MX6Q_ENET_OD_PAD_CTRL	(PAD_CTL_PKE | PAD_CTL_PUE |	\
+		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |	\
+		PAD_CTL_DSE_40ohm | PAD_CTL_HYS | PAD_CTL_ODE)
+
 #define MX6Q_ENET_REF_CLK_PAD_CTRL	(PAD_CTL_PUE |	\
 		PAD_CTL_PUS_100K_UP | PAD_CTL_SPEED_MED |	\
 		PAD_CTL_DSE_40ohm | PAD_CTL_SRE_FAST)
@@ -5214,7 +5218,7 @@
 		(_MX6Q_PAD_ENET_MDIO__RESERVED_RESERVED | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_ENET_MDIO__ENET_MDIO		\
 		(_MX6Q_PAD_ENET_MDIO__ENET_MDIO | \
-		 MUX_PAD_CTRL(MX6Q_ENET_PAD_CTRL))
+		 MUX_PAD_CTRL(MX6Q_ENET_OD_PAD_CTRL))
 #define  MX6Q_PAD_ENET_MDIO__ESAI1_SCKR		\
 		(_MX6Q_PAD_ENET_MDIO__ESAI1_SCKR | MUX_PAD_CTRL(MX6Q_ESAI_PAD_CTRL))
 #define  MX6Q_PAD_ENET_MDIO__SDMA_DEBUG_BUS_DEVICE_3		\
-- 
1.7.9.5

